modules/menus/bluetooth/devices/devicelist.ts

```diff
@@ -101,10 +104,53 @@ const devices = (bluetooth: Bluetooth, self: Box<Gtk.Widget, unknown>): Box<Chil
+                                                                                    : 'Paired',
+                                                                            }),
+                                                                        ];
+                                                                    }
+
+                                                                    return [
+                                                                        Widget.Label({
+                                                                            hpack: 'start',
+                                                                            class_name: 'connection-status dim',
+                                                                            label: device.connected
+                                                                                ? 'Connected'
+                                                                                : 'Paired',
+                                                                        }),
+                                                                        Widget.Separator({
+                                                                            class_name: 'menu-separator',
+                                                                        }),
+                                                                        Widget.Label({
+                                                                            class_name: 'connection-status txt-icon',
+                                                                            label:
+                                                                                device.connected &&
+                                                                                device.battery_percentage !== null
+                                                                                    ? `${batIcon}`
+                                                                                    : '',
+                                                                        }),
+                                                                        Widget.Label({
+                                                                            class_name: 'connection-status battery',
+                                                                            label:
+                                                                                device.connected &&
+                                                                                device.battery_percentage !== null
+                                                                                    ? `${device.battery_percentage}%`
+                                                                                    : '',
+                                                                        }),
```

Comment:
I think a safe bet here then if we don't know for certain if battery percentage exists is to add a safe conditional like
```ts
!isNaN(device.battery_percentage) && device.battery_percentage > 0 
    ? `${device.battery_percentage}%`
    : '',
```

Reply:
We can add this everywhere we rely on the battery to be shown. Also we need to make sure that we don't generate the divider if no battery is present (using the above conditional).

Reply:
lets make a function for this check. 

````
typeof device.battery_percentage === 'number' && device.battery_percentage >= 0 
    ? `${device.battery_percentage}%`
    : '',
```

And add to divider, battery icon and battery percent label

Reply:
is there a way we could perhaps group the divider, battery icon and battery percent label. Then we can apply this condition to group rather than on individual items as we are doing now.

Reply:
We can just put the entire battery related widget set in a box and make the box generate empty children if the battery for that item doesn't exist. Ex:
```ts
Widget.Box({
children: if-battery-exists ? [render the separator, label, label] : []
})
```

Reply:
will try this. This will make the code a lot cleaner